apps/backend/src/app/api/v1/auth/passkey/sign-in/verification-code-handler.tsx

```diff
@@ -80,29 +82,35 @@ export const passkeySignInVerificationCodeHandler = createVerificationCodeHandle
 
     if (!isLocalhost) {
       if (!project.config.domains.map(e => e.domain).includes(parsedOrigin.origin)) {
-        throw new KnownErrors.PasskeyAuthenticationFailed();
+        throw new KnownErrors.PasskeyAuthenticationFailed("Passkey authentication failed because the origin is not allowed");
       } else {
         expectedRPID = parsedOrigin.hostname;
         expectedOrigin = origin;
       }
     }
 
-    const authVerify = await verifyAuthenticationResponse({
-      response: authenticationResponse,
-      expectedChallenge: challenge,
-      expectedOrigin,
-      expectedRPID,
-      credential: {
-        id: passkey.userHandle,
-        publicKey: new Uint8Array(Buffer.from(passkey.publicKey, 'base64')),
-        counter: passkey.counter,
-      },
-      requireUserVerification: false,
-    });
+    let authVerify;
+    try {
+      authVerify = await verifyAuthenticationResponse({
+        response: authentication_response,
+        expectedChallenge: challenge,
+        expectedOrigin,
+        expectedRPID,
+        credential: {
+          id: passkey.userHandle,
+          publicKey: new Uint8Array(Buffer.from(passkey.publicKey, 'base64')),
+          counter: passkey.counter,
+        },
+        requireUserVerification: false,
+      });
+    } catch (error) {
+      captureError("Failed to verify passkey authentication response", error);
+      throw new KnownErrors.PasskeyAuthenticationFailed("Failed to verify passkey authentication response");
+    }
 
 
     if (!authVerify.verified) {
-      throw new StackAssertionError("Passkey authentication failed", { authVerify });
+      throw new KnownErrors.PasskeyAuthenticationFailed("Failed to verify passkey authentication response");
```

Comment:
hmm good question, this happens when something is wrong with the response, but not the credentialID, as that one is "validated" before by trying to find the passkey in the db. This will probably trigger when something is wrong with the challenge or some other field is not encoded properly or some crypto stuff fails.

Reply:
This happens when the signature check fails, but the rest (e.g. formatting of the response and types) were correct, see https://github.com/MasterKale/SimpleWebAuthn/blob/master/packages/server/src/authentication/verifyAuthenticationResponse.ts#L240C21-L240C36